irods/access.py

```diff
@@ -1,5 +1,42 @@
+import collections
+
 class iRODSAccess(object):
 
+    codes = { key:value for key,value in dict(  # copied from iRODS server code:
+        null                 = 1000,            # server/core/include/irods/catalog_utilities.hpp
+        execute              = 1010,
+        read_annotation      = 1020,
+        read_system_metadata = 1030,
+        read_metadata        = 1040,
+        read_object          = 1050,
+        write_annotation     = 1060,
+        create_metadata      = 1070,
+        modify_metadata      = 1080,
+        delete_metadata      = 1090,
+        administer_object    = 1100,
+        create_object        = 1110,
+        modify_object        = 1120,
+        delete_object        = 1130,
+        create_token         = 1140,
+        delete_token         = 1150,
+        curate               = 1160,
+        own                  = 1200
+    ).items()
+    if key in (         # copied from ichmod help text
+        'own',
+        'delete_object',
+        'write', 'modify_object',
+        'create_object',
+        'delete_metadata',
+        'modify_metadata',
+        'create_metadata',
+        'read', 'read_object',
+        'read_metadata',
+        'null')
+    }
```

Comment:
This is being done this way for bridging the write/modify_object and read/read_object gap, right? Just checking

Don't need to do this, but I think an extra indent and/or putting the if on the same line as the `items` line could help with readability here. I keep thinking the if line is the beginning of a new statement, but it's actually a continuation of the `dict` construction.
```suggestion
```suggestion
    codes = {
        key:value for key,value in dict(
            # copied from iRODS server code:
            # server/core/include/irods/catalog_utilities.hpp
            null                 = 1000,
            execute              = 1010,
            read_annotation      = 1020,
            read_system_metadata = 1030,
            read_metadata        = 1040,
            read_object          = 1050,
            write_annotation     = 1060,
            create_metadata      = 1070,
            modify_metadata      = 1080,
            delete_metadata      = 1090,
            administer_object    = 1100,
            create_object        = 1110,
            modify_object        = 1120,
            delete_object        = 1130,
            create_token         = 1140,
            delete_token         = 1150,
            curate               = 1160,
            own                  = 1200
        ).items() if key in (
            'own',
            'delete_object',
            'write', 'modify_object',
            'create_object',
            'delete_metadata',
            'modify_metadata',
            'create_metadata',
            'read', 'read_object',
            'read_metadata',
            'null')
    }
```

Reply:
I certainly had a similar reaction. I had to really stare at the code to figure out what was happening.

My thought was ... would it be clearer to define the mapping explicitly? Keep in mind, this is a question. I'm not saying we must change this.

The primary thing to consider IMO is, how long will it take future "you" to understand it after a year?

Reply:
I see your point. I was just playing with the notion of only using more or less exactly a cut paste of the source of truth .  A testament certainly to pythons flexibility. But i don't mind making it more explicit

Reply:
ok, done